core/src/test/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java

```diff
@@ -469,6 +469,7 @@ public void respond(ActionListener<Response> listener) {
     }
 
     class IndexingAction extends ReplicationAction<IndexRequest, IndexRequest, IndexResponse> {
+        private IndexResponse response;
```

Comment:
I don't like that we have to use this inner state. Can't we move this class to use the bulk with single item infra and keep it as aligned as possible to normal indexing?